### PR TITLE
New simplified API

### PR DESCRIPTION
**Pull request by mmjee**
# Pull Request Template

Part of [AR-3563](https://team-1624093970686.atlassian.net/browse/AR-3563).

## Blocking dependencies

N/A

## Changes

Implemented a new simplified API.

## Checklist

- [x] The branch name follows the format: `developer/AR-XXX-issue-name`.
- [x] The changes have been tested locally.

### CODE REVIEW

**Comment by mmjee** [comment]
The CI failures are unrelated by the way. It happens on “dev” too.

**Comment by SauravKanchan** [comment]
Can you update the usages.md?

You can mention the difference between old and new
## Upload
#### v0.2
```
const arcanaInstance = new arcana.storage.StorageProvider({ appId, provider, email });
const Uploader = arcanaInstance.getUploader();
Uploader.upload(file);
```
#### v0.3
```
const arcanaInstance = new arcana.storage.StorageProvider({ appId, provider, email });
const Uploader = arcanaInstance.upload(file);
```

**Comment by makylfang** [comment]
> const arcanaInstance = new arcana.storage.StorageProvider({ appId, provider, email });
> const did = arcanaInstance.upload(file);

Can we also fix the naming in the docs?
Make it like `storage` instead of `arcanaInstance`.
Also add the importing.
Like 
```js
const { StorageProvider } = window.arcana.storage;
// or
import { StorageProvider } from '@arcana/storage'

const storage = new StorageProvider({ appId, provider, email });
const did = await storage.upload(file);
```



Also @SauravKanchan @mmjee  This is for upload and download. There will be changes to other functions also right? For `share` `revoke` file listing etc.

**Comment by mmjee** [comment]
Adjusted documentation.

P.S. @makylfang Yes, I'll follow up with new PRs for other methods…

File: src/index.ts [hunk 1]
```diff
@@ -295,4 +295,21 @@ export class StorageProvider {
...
     return await makeTx(this.appAddress, this.api, this.provider  , 'linkNFT', [fileId, tokenId, nftContract,chainId]);
   }
+
+  upload = async (fileRaw: any, onProgress: (bytesUploaded: number, bytesTotal: number) => void): Promise<string> => {
+    const uploader = await this.getUploader();
+    uploader.onProgress = onProgress;
+
+    return new Promise((resolve, reject) => {
+      uploader.onError = (err) => reject(err);
+      uploader.upload(fileRaw).then((did) => resolve(did)).catch(e => reject(e));
```
Comment (ajithranka) [inline src/index.ts:305]: Just a suggestion.

```suggestion
      uploader.onError = reject;
      uploader.upload(fileRaw).then(resolve).catch(reject);
```

File: usage.md [hunk 2]
```diff
@@ -6,31 +6,47 @@
...
+## Upload a File object
 
-```typescript
-const Uploader = arcanaInstance.getUploader();
-// file: Blob format
+The `file` object must be an instance of a `Blob` or a descendant (`File`, etc.).
+
+#### v0.3 (for simpler use cases)
+```javascript
+storage.upload(file, (bytesUploaded, bytesTotal) => { console.log('Progress:', (bytesUploaded / bytesTotal).toFixed(2), '%')});
```
Comment (ajithranka) [inline usage.md:20]: Can we expand this code snippet to include handling success and error?

**Review by ajithranka — APPROVED** [review]

File: src/index.ts (hunk 1 above)
Comment (mmjee) [reply src/index.ts]: Added via https://github.com/arcana-network/storage/pull/102/commits/481ecd77015b2d1bdf5cb2ae03ca0c53ebbde207 .

File: usage.md (hunk 2 above)
Comment (mmjee) [reply usage.md]: Done.

File: usage.md [hunk 3]
```diff
@@ -6,31 +6,50 @@
...
 
-## Get Downloader
+## Download from a DID
 
-```typescript
-const Downloader = arcanaInstance.getDownloader();
-// did: DID of file which you want to download
+#### v0.3 (for simpler use cases)
+```javascript
+storage.download(did, (bytesDownloaded, bytesTotal) => { console.log('Progress:', (bytesDownloaded / bytesTotal).toFixed(2), '%')});
```
Comment (makylfang) [inline usage.md:37]: Have to do * 100 to get 0-100 progress.

File: src/index.ts [hunk 4]
```diff
@@ -295,4 +295,21 @@ export class StorageProvider {
...
+      uploader.onError = reject;
+      uploader.upload(fileRaw).then(resolve).catch(reject);
+    });
+  }
+
+  download = async (did: any, onProgress: (bytesDownloaded: number, bytesTotal: number) => void): Promise<void> => {
+    const downloader = await this.getDownloader();
+    downloader.onProgress = onProgress;
+
+    return downloader.download(did)
```
Comment (makylfang) [inline src/index.ts:313]: wont this also have `then().catch()` like upload?

Comment (mmjee) [reply src/index.ts:316]: Yes, but I thought the above example was enough to illustrate how it returns a Promise and how to use a Promise.

File: usage.md (hunk 3 above)
Comment (mmjee) [reply usage.md]: Amended.

**Review by makylfang — APPROVED** [review]

**Review by SauravKanchan — APPROVED** [review]

**Pull request merged**